dot/core/service.go

```diff
@@ -394,6 +394,29 @@ func (s *Service) ProcessBlockAnnounceMessage(msg network.Message) error {
 	return nil
 }
 
+// ProcessBlockRequestMessage processes a block request message, returning a block response message
+func (s *Service) ProcessBlockRequestMessage(msg network.Message) error {
+	blockRequest := msg.(*network.BlockRequestMessage)
+
+	latestBlockNum := s.blockState.LatestHeader().Number
+
+	if blockRequest.StartingBlock[0] == 1 {
```

Comment:
we need to also take in to account the `requestedData` of the block request, they could be asking for header, body, message queue, justification, etc. you should use the `blockState.GetBlockData()` function for retrieving the requested data.

also, right now the block data is stored all at one key in the db, which might make it more complicated to get one specific field. #596 might be relevant here